agent/bench-scripts/pbench-linpack

```diff
@@ -108,48 +120,131 @@ while true; do
+Data alignment value (in Kbytes)            : 4    
+
+Maximum memory requested that can be used=3202964416, at the size=20000
+
+=================== Timing linear equation system solver ===================
+
+Size   LDA    Align. Time(s)    GFlops   Residual     Residual(norm) Check
+20000  20016  4      125.889    ${RANDOM::2}.${RANDOM::4}  2.659155e-10 2.353935e-02   pass
+
+Performance Summary (GFlops)
+
+Size   LDA    Align.  Average  Maximal
+20000  20016  4       ${RANDOM::2}.${RANDOM::4}  42.3718 
+
+Residual checks PASSED
+EOF
+    echo "threads=$2" > "$1/linpack.meta"
+}
+
 ## Ensure the right version of the benchmark is installed
 check_install_rpm "${benchmark_rpm}" "${ver}"
 
 benchmark_fullname="${benchmark}_${config}_${date_suffix}"
 export benchmark_run_dir="$pbench_run/${benchmark_fullname}"
-benchmark_summary_txt_file="$benchmark_run_dir/$benchmark-summary.txt"
-benchmark_summary_html_file="$benchmark_run_dir/$benchmark-summary.html"
 benchmark_iterations="${benchmark_run_dir}/.iterations"
 
 mkdir -p ${benchmark_run_dir}/.running
 
 # now that the benchmark_run_dir directory exists, we can initialize the iterations file
 > ${benchmark_iterations}
 
-iteration=1-$threads-threads
-sample="sample1"
-echo $iteration >> $benchmark_iterations
-benchmark_results_dir="$benchmark_run_dir/$iteration/${sample}"
-result_file=$benchmark_results_dir/result.txt
-mkdir -p $benchmark_results_dir
-
 # Start the tool meisters on each registered local/remote host
+echo "Starting tool-meister"
 pbench-tool-meister-start --sysinfo="${sysinfo}" "${tool_group}"
 if [[ ${?} != 0 ]]; then
-	error_log "[${script_name}]: failed to start the tool meisters."
-	exit 1
+    error_log "[${script_name}]: failed to start the tool meisters."
+    exit 1
 fi
 
 trap "interrupt" INT QUIT TERM
 
-## Run the benchmark and start/stop perf analysis tools
-pbench-start-tools --group=$tool_group --dir=$benchmark_results_dir
-
-OMP_NUM_THREADS=$threads ${linpack_dir}/${linpack_cmd} < ${linpack_dir}/${linpack_dat} | tee $result_file
-
-pbench-stop-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-send-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-postprocess-tools --group=$tool_group --dir=$benchmark_results_dir
-
-ln -s ${sample} $benchmark_run_dir/$iteration/reference-result
+count=1
+for thread in ${threads//,/ }; do
+    iteration="$count-$thread-threads"
+    echo $iteration >> $benchmark_iterations
+    for sample in `seq 1 $nr_samples`; do
+        echo "Starting sample $iteration/sample$sample"
+        benchmark_results_dir="$benchmark_run_dir/$iteration/sample$sample"
+        mkdir -p $benchmark_results_dir
+
+        # Generate the linpack.input dirs (in /var/lib/pbench-agent/tmp)
+        store_and_run "$benchmark_results_dir" "'$pbench_bin/bench-scripts/postprocess/linpack-prepare-input-file' --threads '$thread' --run-samples 1 --linpack-binary '$linpack_dir/$linpack_cmd'" "$benchmark-preprocess.cmd"
```

Comment:
Seems like `linpack-prepare-input-file` needs to be added to this commit?

Reply:
`agent/bench-scripts/postprocess/linpack-prepare-input-file` is an existing file in the Pbench repo...are you expecting changes to it?

Reply:
I was expecting it to be renamed to what the other bench-scripts do, but using it as it stands is okay.